musica-fortran/test/test_musica_api.F90

```diff
@@ -1,38 +1,44 @@
 program test
     use iso_c_binding
-    use musica_micm
-    implicit none
-    type(micm_t) :: m
+    use micm_core
 
-    real(c_double) :: temperature      
-    real(c_double) :: pressure
-    real(c_double) :: time_step
-    real(c_double), dimension(10) :: concentrations
-    integer(c_size_t) :: num_concentrations
+    implicit none
+    
+    type(micm_t)                   :: micm
+    real(c_double)                 :: temperature      
+    real(c_double)                 :: pressure
+    real(c_double)                 :: time_step
+    integer                        :: num_concentrations
+    real(c_double), dimension(10)  :: concentrations
+    integer                        :: errcode
 
     temperature = 10d0
     pressure = 20d0
     time_step = 1d0
-    concentrations = (/ 1d0, 2d0, 3d0, 4d0, 5d0, 6d0, 7d0, 8d0, 9d0, 10d0 /)
     num_concentrations = 10
+    concentrations = (/ 1d0, 2d0, 3d0, 4d0, 5d0, 6d0, 7d0, 8d0, 9d0, 10d0 /)
+
+    write(*,*) "  * [Fortran] Creating MICM..."
+    micm = micm_t("micm_config")
 
-    write(*,*) "  * [Fortran] Creating MICM"
-    m = micm_t("micm_config")
+    write(*,*) "  * [Fortran] Creating solver..."
+    errcode = micm%create_solver()
 
-    write(*,*) "  * [Fortran] Creating solver"
-    write(*,*) "  * [Fortran] Solver creating status indicates ", m%create_solver(), " (1 is success, else failure) "
+    if (errcode == 1) then
+        write(*,*) "  * [Fortran] Failed in creating solver"
```

Comment:
Sorry, I missed this earlier - if this test is expected to successfully create a solver, I would add:
```
stop 3
```
to the block that catches failures creating the solver. This way the test will fail if there is a problem creating the solver.